Build failure against TF r2.16: third_party/eigen3/unsupported/Eigen/CXX11/Tensor: No such file or directory

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 20.04): Alma Linux 9.3
- TensorFlow version and how it was installed (source or binary): r2.16 branch from source
- TensorFlow-Addons version and how it was installed (source or binary): master branch from source
- Python version: 3.11
- Is GPU used? (yes/no): yes

**Describe the bug**

Build fails with:

```
In file included from tensorflow_addons/custom_ops/seq2seq/cc/kernels/beam_search_ops.cc:22:
./tensorflow_addons/custom_ops/seq2seq/cc/kernels/beam_search_ops.h:21:10: fatal error: third_party/eigen3/unsupported/Eigen/CXX11/Tensor: No such file or directory
   21 | #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

A clear and concise description of what the bug is.

**Code to reproduce the issue**

```
git clone https://github.com/tensorflow/addons.git
cd addons
git checkout master
python3 ./configure.py
bazel build build_pip_pkg
```

--- a/tensorflow_addons/custom_ops/image/cc/kernels/adjust_hsv_in_yiq_op.h
+++ b/tensorflow_addons/custom_ops/image/cc/kernels/adjust_hsv_in_yiq_op.h
@@ -23,8 +23,14 @@ limitations under the License.
 #include "tensorflow/core/framework/op_kernel.h"
 #include "tensorflow/core/framework/register_types.h"
 #include "tensorflow/core/framework/types.h"
+#include "tensorflow/core/public/version.h"
+#if TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 16
+#include "Eigen/Core"
+#include "unsupported/Eigen/CXX11/Tensor"
+#else
 #include "third_party/eigen3/Eigen/Core"
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#endif
 
 namespace tensorflow {
 namespace addons {

--- a/tensorflow_addons/custom_ops/image/cc/kernels/connected_components.h
+++ b/tensorflow_addons/custom_ops/image/cc/kernels/connected_components.h
@@ -26,8 +26,13 @@ limitations under the License.
 #include "tensorflow/core/framework/op_kernel.h"
 #include "tensorflow/core/framework/tensor_types.h"
 #include "tensorflow/core/platform/types.h"
+#include "tensorflow/core/public/version.h"
 #include "tensorflow/core/util/work_sharder.h"
+#if TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 16
+#include "unsupported/Eigen/CXX11/Tensor"
+#else
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#endif
 
 namespace tensorflow {
 namespace addons {

--- a/tensorflow_addons/custom_ops/image/cc/kernels/euclidean_distance_transform_op.h
+++ b/tensorflow_addons/custom_ops/image/cc/kernels/euclidean_distance_transform_op.h
@@ -21,7 +21,12 @@ limitations under the License.
 #include "tensorflow/core/framework/op_kernel.h"
 #include "tensorflow/core/framework/tensor_types.h"
 #include "tensorflow/core/platform/types.h"
+#include "tensorflow/core/public/version.h"
+#if TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 16
+#include "unsupported/Eigen/CXX11/Tensor"
+#else
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#endif
 
 namespace tensorflow {
 namespace addons {

--- a/tensorflow_addons/custom_ops/seq2seq/cc/kernels/beam_search_ops.cc
+++ b/tensorflow_addons/custom_ops/seq2seq/cc/kernels/beam_search_ops.cc
@@ -32,8 +32,13 @@ limitations under the License.
 #include "tensorflow/core/framework/types.h"
 #include "tensorflow/core/platform/logging.h"
 #include "tensorflow/core/platform/macros.h"
+#include "tensorflow/core/public/version.h"
 #include "tensorflow/core/util/work_sharder.h"
+#if TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 16
+#include "unsupported/Eigen/CXX11/Tensor"
+#else
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#endif
 
 namespace tensorflow {
 namespace addons {

--- a/tensorflow_addons/custom_ops/seq2seq/cc/kernels/beam_search_ops.h
+++ b/tensorflow_addons/custom_ops/seq2seq/cc/kernels/beam_search_ops.h
@@ -18,7 +18,12 @@ limitations under the License.
 
 #include "tensorflow/core/framework/tensor_types.h"
 #include "tensorflow/core/platform/types.h"
+#include "tensorflow/core/public/version.h"
+#if TF_MAJOR_VERSION == 2 && TF_MINOR_VERSION >= 16
+#include "unsupported/Eigen/CXX11/Tensor"
+#else
 #include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#endif
 
 namespace tensorflow {
 class OpKernelContext;

